src/tasks/delete-item-tags-by-item-id-task.ts

```diff
@@ -0,0 +1,33 @@
+// global
+import { DatabaseTransactionHandler, ItemMembershipService, ItemService, Member } from 'graasp';
+import { TAGS } from '../constants';
+import { ItemTagService } from '../db-service';
+import { BaseItemTagTask } from './base-item-tag-task';
+
+export class DeleteItemTagsByItemIdTask extends BaseItemTagTask<Member, number> {
+  get name(): string {
+    return DeleteItemTagsByItemIdTask.name;
+  }
+
+  private itemId: string;
+
+  constructor(
+    member: Member,
+    itemId: string,
+    itemService: ItemService,
+    itemMembershipService: ItemMembershipService,
+    itemTagService: ItemTagService,
+  ) {
+    super(member, itemTagService, itemService, itemMembershipService);
+    this.itemId = itemId;
+  }
+
+  async run(handler: DatabaseTransactionHandler): Promise<void> {
+    this.status = 'RUNNING';
+    const itemPath = this.itemId.replace(/-/g, '_');
+    const tags = (await this.itemTagService.getAllTags(handler))?.filter((tag) => tag?.name === TAGS.PUBLIC || tag?.name === TAGS.PUBLISHED);
```

Comment:
There's a naming conflict between the name of the task and the fact that you actually deleting specific tags. Either rename the task so it contains "public" and "publish" in the name, or have a generic task where you pass the tagid (this would be the best solution).